nemo/collections/nlp/modules/common/megatron/megatron_perceiver_encoders.py

```diff
@@ -0,0 +1,271 @@
+            relative_attention_max_distance=self.relative_attention_max_distance,
+            use_cpu_initialization=self.use_cpu_initialization,
+            bias_gelu_fusion=self.bias_gelu_fusion,
+            bias_dropout_fusion=self.bias_dropout_add_fusion,
+            masked_softmax_fusion=self.masked_softmax_fusion,
+            persist_layer_norm=self.persist_layer_norm,
+            openai_gelu=self.openai_gelu,
+            onnx_safe=self.onnx_safe,
+            activation=self.activation,
+            bias=self.bias,
+            normalization=self.normalization,
+            model_type=self.parent_model_type,
+            transformer_block_type=self.transformer_block_type,
+            headscale=self.headscale,
+        )
+
+    def _build_self_attn_layer(self):
+        return ParallelTransformer(
+            layer_type=LayerType.encoder,
+            init_method=self.init_method,
+            output_layer_init_method=self.output_layer_init_method,
+            num_layers=1,
+            hidden_size=self.hidden_size,
+            num_attention_heads=self.num_attention_heads,
+            apply_query_key_layer_scaling=self.apply_query_key_layer_scaling,
+            kv_channels=self.kv_channels,
+            ffn_hidden_size=self.ffn_hidden_size,
+            self_attn_mask_type=self.model_attn_mask_type,
+            pre_process=self.pre_process,
+            post_process=False,  # This is to avoid the final layernorm and transpose.
+            precision=self.precision,
+            fp32_residual_connection=self.fp32_residual_connection,
+            activations_checkpoint_method=self.activations_checkpoint_method,
+            activations_checkpoint_num_layers=self.activations_checkpoint_num_layers,
+            layernorm_epsilon=self.layernorm_epsilon,
+            hidden_dropout=self.hidden_dropout,
+            attention_dropout=self.attention_dropout,
+            position_embedding_type=self.position_embedding_type,
+            relative_attention_num_buckets=self.relative_attention_num_buckets,
+            relative_attention_max_distance=self.relative_attention_max_distance,
+            use_cpu_initialization=self.use_cpu_initialization,
+            bias_gelu_fusion=self.bias_gelu_fusion,
+            bias_dropout_fusion=self.bias_dropout_add_fusion,
+            masked_softmax_fusion=self.masked_softmax_fusion,
+            persist_layer_norm=self.persist_layer_norm,
+            openai_gelu=self.openai_gelu,
+            onnx_safe=self.onnx_safe,
+            activation=self.activation,
+            bias=self.bias,
+            normalization=self.normalization,
+            model_type=self.parent_model_type,
+            transformer_block_type=self.transformer_block_type,
+            headscale=self.headscale,
+        )
+
+    def set_input_tensor(self, input_tensor):
+        """ See megatron.model.transformer.set_input_tensor()"""
+        # TODO: Fix this.
+        pass
```

Comment:
This is required only for PP right? If so, can you add that to the comment?

Reply:
Done.